develop/dev-guide-proxysql-integration.md

```diff
@@ -0,0 +1,677 @@
+
+```shell
+./proxy-rule-split.sh
+```
+
+#### Run step by step
+
+The preceding `proxy-rule-split.sh` script can be run step by step as follows:
+
+1. Start two TiDB containers and a ProxySQL instance.
+
+    ```shell
+    docker-compose up -d
+    ```
+
+    - Start two TiDB containers using `docker-compose`. All the ports in the container are `4000` and host ports are `4001` and `4002`.
+    - After starting TiDB containers, the ProxySQL instance is started. The port of **_ProxySQL MySQL Interface_** in container is `6033` and the host port is `6034`.
+    - The port of **_ProxySQL Admin Interface_** is not exposed because it can only be accessed in the container.
+    - For more details about the process, refer to [`docker-compose.yaml`](https://github.com/Icemap/tidb-proxysql-integration-test/blob/main/example/proxy-rule-admin-interface/docker-compose.yaml)
+
+2. In the two TiDB containers, create the same table schema with different data (`'tidb-0'` and `'tidb-1'`) to distinguish TiDB instances.
+
+    ```shell
+    mysql -u root -h 127.0.0.1 -P 4001 << EOF
+    DROP TABLE IF EXISTS test.test;
+    CREATE TABLE test.test (db VARCHAR(255));
+    INSERT INTO test.test (db) VALUES ('tidb-0');
+    EOF
+
+    mysql -u root -h 127.0.0.1 -P 4002 << EOF
+    DROP TABLE IF EXISTS test.test;
+    CREATE TABLE test.test (db VARCHAR(255));
+    INSERT INTO test.test (db) VALUES ('tidb-1');
+    EOF
+    ```
+
+3. To execute the [`proxysql-prepare.sql`](https://github.com/Icemap/tidb-proxysql-integration-test/blob/main/example/proxy-rule-admin-interface/proxysql-prepare.sql) in **_ProxySQL Admin Interface_**, executing the `docker-compose exec` command as follows:
+
+    ```shell
+    docker-compose exec proxysql sh -c "mysql -uadmin -padmin -h127.0.0.1 -P6032 < ./proxysql-prepare.sql"
+    ```
+
+    The preceding SQL file runs:
+
+    1. Add hosts of two TiDB Servers. The `hostgroup_id` of `tidb-0` is `0` and `hostgroup_id` of `tidb-1` is `1`.
+    2. Make the configuration of TiDb Servers effective and save it on disk.
+    3. Add a user `root` with an empty password and set `default_hostgroup` as `0`. It indicates that the SQL routes to `tidb-0` by default.
+    4. Make the configuration of the user effective and save it on disk.
+    5. Add the rule `^SELECT.*FOR UPDATE$` with `rule_id` as `1` and `destination_hostgroup` as `0`. If a SQL statement match this rule, it used the TiDB Server with `hostgroup` as `0` (this rule forwards `SELECT ... FOR UPDATE` to the written database).
+    6. Add the rule `^SELECT` with `rule_id` as `2` and `destination_hostgroup` as `1`. If a SQL statements match this rule, it uses the TiDB Server with `hostgroup` as `1`.
+    7. Make the configuration of rule effective and save it on disk.
```

Comment:
ditto, who is taking these steps, the user or the SQL script?

Reply:
SQL Script.

Reply:
If so, please update these lines similar to my previous comments.

Reply:
Roger that.